src/components/OSCALControlImplementation.js

```diff
@@ -52,26 +65,37 @@ export default function OSCALControlImplementation(props) {
             </Grid>
             <Grid item xs={12}>
               <List className={classes.OSCALControlImplementationImplReqList}>
-                {props.controlImplementation["implemented-requirements"].map(
-                  (implementedRequirement) => (
-                    <OSCALControlImplementationImplReq
-                      implementedRequirement={implementedRequirement}
-                      components={props.components}
-                      controls={props.controls}
-                      childLevel={0}
-                      key={implementedRequirement.uuid}
-                      modifications={props.modifications}
-                      isEditable={props.isEditable}
-                      onRestSuccess={props.onRestSuccess}
-                      onRestError={props.onRestError}
-                      partialRestData={props.partialRestData}
-                    />
-                  )
-                )}
+                {implementedRequirements.map((implementedRequirement) => (
+                  <OSCALControlImplementationImplReq
+                    implementedRequirement={implementedRequirement}
+                    components={props.components}
+                    controls={props.controls}
+                    childLevel={0}
+                    key={implementedRequirement.uuid}
+                    modifications={props.modifications}
+                    isEditable={props.isEditable}
+                    onRestSuccess={props.onRestSuccess}
+                    onRestError={props.onRestError}
+                    partialRestData={props.partialRestData}
+                  />
+                ))}
               </List>
             </Grid>
           </Grid>
         </CardContent>
+        {props.isEditable ? (
+          <Grid item className={classes.OSCALControlImplementationAdd}>
+            <OSCALControlImplementationAdd
+              controls={props.controls}
+              implementedControls={controlIds}
+              implementedRequirements={implementedRequirements}
+              onRestSuccess={props.onRestSuccess}
+              onRestError={props.onRestError}
+              partialRestData={props.partialRestData}
+              setImplementedRequirements={setImplementedRequirements}
```

Comment:
Is this only used to display the new implemented requirements while the request is processing?  If so, perhaps we should just display circular progress like the editing existing implementation?

Reply:
This is used to allow a user to view the implemented requirement prior to saving.

Reply:
We should consider simply making the REST request on click of what is currently the check button (changing that icon to save).

I think that change would lead to a better UX and simpler implementation.